portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,29 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> messages;
```

Comment:
Why do you need messages as private field? This creates problems if several calls made to doGet made concurrently.

Reply:
I was taught that it's good convention to have instance fields as private variables. Would it be better to just initialize it after the declaration and add all the messages to it, outside of the doGet method?

Reply:
Class members should be created as last resort and only when represent important object state. 
I do not see need for this property.

Reply:
I'm still a bit confused on how it's not an important object state and how it creates problems to the doGet method. Could you explain?

Reply:
1. If you remove messages as member and then change line 40:
  new ArrayLisat<String> messah = new ArrayList<>();
then your code will still works without messages as member.
Your messages is just interim data.. 

2. I would question if you really need messages at all. 
Your code first prepare array list and then convert to json object. 
You can directly prepare json object without interim structure.

Reply:
Okay, I just fixed it. I might end up deleting this whole section with the messages and getMessages, because those were the hardcoded introduction for listing people's comments that they submit on my website.